src/app/_components/MainLayoutProviders.tsx

```diff
@@ -22,8 +24,11 @@ export default function MainLayoutProviders({
           <UserProvider
             storageKey="nosgestesclimat::v3"
             migrationInstructions={migrationInstructions}
-            initialRegionCode={regionCode}>
-            <PreventNavigationProvider>{children}</PreventNavigationProvider>
+            initialRegion={initialRegion}>
+            <PreventNavigationProvider>
+              <MainHooks />
+              {children}
```

Comment:
Why don´t you wrap `{children}` in main hooks here ?

I agree `Mainhooks` hooks look synchronous but imagine a day where we need a main hook to asynchronously redirect user or do whatever action. We will render children before the end of the hook no ?

Reply:
It was because MainHooks was in a Suspense before. It is not the case now so you are right, it might be better to render children inside it